tests/ui/implicit_hasher.stderr

```diff
@@ -0,0 +1,154 @@
+error: impl for `HashMap` should be generarized over different hashers
+  --> $DIR/implicit_hasher.rs:11:35
+   |
+11 | impl<K: Hash + Eq, V> Foo<i8> for HashMap<K, V> {
+   |                                   ^^^^^^^^^^^^^
+   |
+   = note: `-D implicit-hasher` implied by `-D warnings`
+help: consider adding a type parameter
+   |
+11 | impl<K: Hash + Eq, V, S: ::std::hash::BuildHasher + Default> Foo<i8> for HashMap<K, V> {
+   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+help: ...and change the type to
+   |
+11 | impl<K: Hash + Eq, V> Foo<i8> for HashMap<K, V, S> {
+   |                                   ^^^^^^^^^^^^^^^^
+help: ...and use generic constructor here
+   |
+14 |         let _: HashMap<i32, i32> = HashMap::default();
+   |                                    ^^^^^^^^^^^^^^^^^^
+help: ...and use generic constructor here
+   |
+15 |         let _: HashSet<i32> = HashSet::default();
+   |                               ^^^^^^^^^^^^^^^^^^
+help: ...and use generic constructor here
+   |
+17 |         (HashMap::default(), HashMap::with_capacity(10))
+   |          ^^^^^^^^^^^^^^^^^^
+help: ...and use generic constructor here
+   |
+17 |         (HashMap::new(), HashMap::with_capacity_and_hasher(10, Default::default()))
```

Comment:
You can use https://github.com/rust-lang-nursery/rust-clippy/blob/73a1dd8e7f10f1c6cba618b60e3fed690a72a8d7/clippy_lints/src/utils/mod.rs#L667 to merge all suggestions into one. That should merge this one and the above one into a single one.